src/KBarSearch.tsx

```diff
@@ -33,12 +37,12 @@ export function KBarSearch(
   }, [currentRootActionId, query]);
 
   const placeholder = React.useMemo(
-    () =>
-      currentRootActionId
+    (): string => {
+      const text =  props.defaultPlaceholder ?? "Type a command or search…";
+      return currentRootActionId
         ? actions[currentRootActionId].name
-        : "Type a command or search…",
-    [actions, currentRootActionId]
-  );
+        : text;
```

Comment:
```suggestion
        : default;
```

Reply:
I'm not able to use default as the name of the value because it's a keyword, but I applied all the other changes.